tests/templates/utils/matcher.test.ts

```diff
@@ -0,0 +1,212 @@
+import { describe, test, expect } from 'vitest';
+import { parse } from 'cookie';
+import { hasField, checkRouteMatch } from '../../../templates/_worker.js/utils';
+
+type HasFieldTestCase = {
+	name: string;
+	has: VercelHasFields[0];
+	expected: boolean;
+};
+
+type CheckRouteMatchTestCase = {
+	name: string;
+	route: VercelSource;
+	requiredStatus?: number;
+	expected: boolean;
+};
+
+const req = new Request('https://test.com/index?foo=bar&bar=', {
+	headers: {
+		headerWithValue: 'value',
+		headerWithoutValue: undefined as unknown as string,
+		cookie: 'foo=bar; bar=',
```

Comment:
I would suggest using the same convention you used for the headers:
```suggestion
		cookie: 'cookieWithValue=value; cookieWithoutValue=',
```
(I especially dislike the double use of `bar` here since in the tests when you will see `bar` you might get confused to which one the test is referring to)